oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java

```diff
@@ -262,7 +274,7 @@ public int getProcessCount() {
 
     @Override
     public int getThreadId() {
-        return LinuxLibc.INSTANCE.gettid();
+        return HAS_GETTID ? LinuxLibc.INSTANCE.gettid() : LinuxLibc.INSTANCE.syscall(LinuxLibc.SYS_GETTID).intValue();
```

Comment:
Can we do one more fallback check here?  Collect this return value and if it's 0, then use  the `/proc/thread-self` symbolic link name.

Reply:
Under the redhat linux not have /proc/thread-self
ls /proc/thread-self
ls: cannot access /proc/thread-self: No such file or directory

Reply:
Well the syscall should work there so it doesn't get to that third fallback.  That one is necessary on Alpine Linux though.